### PR TITLE
feat: make actors reschedule indefinitely with backoff

### PR DESCRIPTION
**Pull request by MasterPtato**
Fixes RVT-4196
Fixes RVT-4179

**Implemented but untested**

### CODE REVIEW

**Comment by MasterPtato** [comment]
* **#1475** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1475?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/1475?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by NathanFlurry — CHANGES_REQUESTED** [review]
question on slack regarding retry if actor fails to start vs just retrying on failed alloc

edit: handle retries for all reschedule events except for draining

**Comment by MasterPtato** [comment]
@NathanFlurry updated to reset retry count only after ready

File: packages/services/ds/src/workflows/server/pegboard/mod.rs [hunk 1]
```diff
@@ -646,27 +695,57 @@ async fn reschedule_actor(
...
+							retry_count.try_into()?,
+						);
+						let next = backoff.step().expect("should not have max retry");
 
-			// Throw the original error from the setup activities
-			return Err(err);
-		}
-	};
+						// Sleep for backoff
+						ctx.sleep_until(next).await?;
```
Comment (NathanFlurry) [inline packages/services/ds/src/workflows/server/pegboard/mod.rs:733]: I think this is still wrong. We talked about also adding backoff for an actor that crashes, so shouldn't the backoff be executed at the start of reschedule_actor? Like `if !first_reschedule { /* do backoff */ }`

Context: https://rivet-gg.slack.com/archives/D02R3KVNKHN/p1732672061669669?thread_ts=1732671085.517149&cid=D02R3KVNKHN

**Review by NathanFlurry — CHANGES_REQUESTED** [review]

File: packages/services/ds/src/workflows/server/pegboard/mod.rs (hunk 1 above)
Comment (MasterPtato) [reply packages/services/ds/src/workflows/server/pegboard/mod.rs:733]: Actors always retry immediately the first time, then backoff kicks in: https://rivet-gg.slack.com/archives/D02R3KVNKHN/p1732672128678809?thread_ts=1732671085.517149&cid=D02R3KVNKHN

**Review by NathanFlurry — APPROVED** [review]

**Pull request closed without merging**